<?xml version="1.0"?>

<!--
	In this launch file all the necessary nodes are started which are needed to accumulate the sensor data from the Hokuyo and the Velodyne scanner. They are published as a sensor_msgs::PointCloud2
-->

<launch>


  	<!-- rosbag is started using simulation time -->
	<param name="use_sim_time" type="boolean" value="true" />
	<node pkg="rosbag" type="play" name="player" args="--clock /home/jannik/Bachelorarbeit/Bagfiles/2020-03-09-14-36-17.bag"/>

	<group ns="mapping/hokuyo_assembler">
	  	<!-- 
	  		sensor data from the Hokuyo comes as sensor_msgs::LaserScan messages on the topic /mascor/SWAP/hokuyo/scan/raw. The laser_scan_assembler collects these messages and transforms them into the frame /mascor/platform_laser_rotation_base.
	  	-->
	  	<node type="laser_scan_assembler" pkg="laser_assembler" name="laser_scan_assembler" output="screen">
	    	<remap from="scan" to="/mascor/SWAP/hokuyo/scan/raw"/>
	    	<param name="max_scans" type="int" value="100000" />
	    	<param name="fixed_frame" type="string" value="mascor/platform_laser_rotation/base" />
	  	</node>

	  	<!--
	  		The laser_assembler_client calls the assemble_scans service of the laser_scan_assembler at 'rate' and all the collected LaserScans are assembled in a sensor_msgs::PointCloud and published on /mapping/assembler/pointcloud_raw.
	  	-->

	  	<node pkg="pc_accumulation" type="laser_assembler_client" name="laser_assembler_client" output="screen">
	  		<remap from="assembled_pointcloud" to="/mapping/assembler/pointcloud_raw"/>
	  		<param name="rate" value="$(eval 1 / 0.3)"/>
	  	</node>
	</group>

	<group ns="mapping/assembler">
	  	<!--
	  		sensor data from the Velodyne comes as sensor_msgs::PointCloud2 messages. They are converted to sensor_msgs::PointCloud by the pointcloud_converter and published on mapping/assembler/pointcloud_raw (same as assembled clouds from hokuyo).
	  	
	  	<node pkg="pc_accumulation" type="pointcloud_converter_node" name="velodyne_pointcloud_converter" output="screen">
	  		<remap from="pointcloud2_in" to="/mascor/SWAP/velodyne/cloud_raw"/>
	  		<remap from="pointcloud_out"  to="pointcloud_raw"/>
	  	</node>
		-->

	  	<!--
			The point_cloud_assembler collects the messages from the topic mapping/assembler/pointcloud_raw and transfroms them into the frame /mascor/platform_laser_rotation_base.
		
	  	<node pkg="laser_assembler" type="point_cloud_assembler" name="point_cloud_assembler" output="screen">
	  		<remap from="cloud" to="pointcloud_raw"/>
	    	<param name="max_clouds" type="int" value="100000" />
	    	<param name="fixed_frame" type="string" value="mascor/platform_laser_rotation/base" />
	    </node>
		-->

	    <!--
	  		The laser_assembler2_client calls the assemble_scans2 service of the laser_scan_assembler at 'rate' and all the collected PointClouds are assembled in a sensor_msgs::PointCloud2 and published on /mapping/assembler/pointcloud.
	  	
	    <node pkg="pc_accumulation" type="laser_assembler2_client" name="laser_assembler2_client" output="screen">
	  		<remap from="rate" to="/mapping/rate"/>
	    </node>
	    -->
	</group>

</launch>
